src/routes/jira/sync/jira-sync-post.ts

```diff
@@ -9,8 +9,12 @@ import { booleanFlag, BooleanFlags } from "~/src/config/feature-flags";
 export const JiraSyncPost = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
 	const { installationId: gitHubInstallationId, appId: gitHubAppId } = req.body;
 
+	if (!req.body?.commitsFromDate) {
+		res.status(404).send("Missing sync date");
+		return;
+	}
 	// A date to start fetching commit history(main and branch) from.
-	const commitsFromDate = req.body.commitsFromDate ? new Date(req.body.commitsFromDate) : undefined;
+	const commitsFromDate = new Date(req.body.commitsFromDate);
```

Comment:
/nit
Ideally the format of the `req.body.commitsFrom` will need validation as well.

Reply:
Date format validation is already done at router level.